machine-learning/app/main.py

```diff
@@ -51,6 +51,9 @@ async def lifespan(_: FastAPI) -> AsyncGenerator[None, None]:
             log.info(f"Initialized request thread pool with {settings.request_threads} threads.")
         if settings.model_ttl > 0 and settings.model_ttl_poll_s > 0:
             asyncio.ensure_future(idle_shutdown_task())
+        if settings.preload != "":
+            log.info(f"Preloading models: {settings.preload}")
+            model_cache.preload_models(settings.preload)
```

Comment:
Thanks a lot for the detailed review! I made these changes regarding `ttl` and I agree that it's a simpler way of doing this, however I now ran into a problem with the `RevalidationPlugin`. 
The way it worked was by updating the `ttl` of the cache entry with the `ttl` of the `SimpleMemoryCache` client, but now that the ttl is passed per `model_cache.get` call we don't have access to it. In theory we could get it directly from the settings, or add it back into the `ModelCache` class but I'm not sure that it's the best way of doing this. It would probably be best if we could be able to set it per-model, but I don't think there's currently a way of checking the set `ttl`, we have access to the `TimerHandle` but that only gives as the absolute time, not the `ttl` set during `model_cache.get`
I also intially worried about revalidating preloaded models but that's not an issue due to them not having `TimerHandle`.

Reply:
Ahh that's a good point. We could move the revalidation to a method in the model cache that gets called before returning in `get`.

Reply:
Alright that's a good idea, but I still see a small issue with this approach. Even though we're loading the models in main with `ttl = None`, once we call `get` we're resetting the `ttl` back to `model_ttl` so the model will not stay loaded. Is it fine to check if model is supposed to be preloaded based on `settings.preload` inside the `predict` function/endpoint?

Reply:
The revalidation plugin checks if there's a timer handle right? The same check would be here as well.

Reply:
Good point, done